components/BottomNavigation/examples/BottomNavigationResetButtons.swift

```diff
@@ -0,0 +1,129 @@
+  let tabBarItem4 = UITabBarItem(title: "Cake", image: UIImage(named: "Cake"), tag: 3)
+
+  let buttonOne = MDCButton()
+  let buttonTwo = MDCButton()
+
+  init() {
+    super.init(nibName: nil, bundle: nil)
+  }
+
+  @available(*, unavailable)
+  required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+  }
+
+  func layoutBottomNavBar() {
+    let size = bottomNavBar.sizeThatFits(view.bounds.size)
+    let bottomNavBarFrame = CGRect(x: 0,
+                                   y: view.bounds.height - size.height,
+                                   width: size.width,
+                                   height: size.height)
+    bottomNavBar.frame = bottomNavBarFrame
+  }
+
+  override func viewWillLayoutSubviews() {
+    super.viewWillLayoutSubviews()
+    layoutBottomNavBar()
+  }
+
+  #if swift(>=3.2)
+  @available(iOS 11, *)
+  override func viewSafeAreaInsetsDidChange() {
+    super.viewSafeAreaInsetsDidChange()
+    layoutBottomNavBar()
+  }
+  #endif
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    view.backgroundColor = .lightGray
+    view.addSubview(bottomNavBar)
+
+    bottomNavBar.alignment = .centered
+
+    // Add items to the bottom navigation bar.
+    bottomNavBar.items = [ tabBarItem1, tabBarItem2, tabBarItem3 ]
+
+    // Select a bottom navigation bar item.
+    bottomNavBar.selectedItem = tabBarItem2
+
+    // Layout buttons
+    buttonOne.setTitle("Reorder One", for: .normal)
+    buttonOne.sizeToFit()
+    buttonOne.frame.origin = CGPoint(x: view.center.x - (buttonOne.frame.width / 2),
+                                     y: view.center.y - (buttonOne.frame.height + 16))
+    buttonOne.addTarget(self, action: #selector(reorderOne), for: .touchUpInside)
+    view.addSubview(buttonOne)
+
+    buttonTwo.setTitle("Reorder Two", for: .normal)
+    buttonTwo.sizeToFit()
+    buttonTwo.frame.origin = CGPoint(x: view.center.x - (buttonTwo.frame.width / 2),
+                                      y: view.center.y + 16)
+    buttonTwo.addTarget(self, action: #selector(reorderTwo), for: .touchUpInside)
+    view.addSubview(buttonTwo)
+
+    MDCBottomNavigationBarColorThemer.applySemanticColorScheme(colorScheme,
+                                                               toBottomNavigation: bottomNavBar)
+  }
+
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    self.navigationController?.setNavigationBarHidden(true, animated: animated)
+  }
+
+  @objc func reorderOne(_ button: UIButton) {
+    bottomNavBar.items = [tabBarItem2, tabBarItem3, tabBarItem1]
+  }
+
+  @objc func reorderTwo(_ button: UIButton) {
+    bottomNavBar.items = [tabBarItem3, tabBarItem1, tabBarItem2, tabBarItem4]
+    bottomNavBar.selectedItem = tabBarItem3
+  }
+}
+
+// MARK: Catalog by convention
+extension BottomNavigationResetExample {
+  class func catalogBreadcrumbs() -> [String] {
+    return ["Bottom Navigation", "Bottom Navigation Reset (Swift)"]
```

Comment:
I'm not sure I understand the "reset" part of this. Isn't it a reorder/reassignment example?